core/app/models/mno_enterprise/base_resource.rb

```diff
@@ -130,9 +130,24 @@ def ==(o)
   end
 end
 
+LocaleMiddleware = Struct.new(:app) do
```

Comment:
Maybe you could extract this to lib file ?

Reply:
Let's keep it in this file for now. I prefer to not scatter the logic too much for one-off usages. If we see ourself using another custom middleware then we'll extract.